x/emissions/keeper/inference_synthesis/synth_palette_weight.go

```diff
@@ -64,23 +65,34 @@ func (p *SynthPalette) CalcWeightsGivenWorkers() (RegretInformedWeights, error)
 		}
 	}
 
+	moreThanOneNotNewInferer := p.SingleInfererNotNew == ""
+
 	// Calculate the weights from the normalized regrets
 	infererWeights := make(map[Worker]Weight)
 	for _, worker := range p.Inferers {
-		infererWeight, err := CalcWeightFromNormalizedRegret(normalizedInfererRegrets[worker], maxRegret, p.PNorm, p.CNorm)
-		if err != nil {
-			return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating inferer weight")
+		// If there is more than one not-new inferer, calculate the weight for the ones that are not new
+		var infererWeight = alloraMath.ZeroDec()
+		if moreThanOneNotNewInferer && !p.InfererRegrets[worker].noPriorRegret {
+			infererWeight, err = CalcWeightFromNormalizedRegret(normalizedInfererRegrets[worker], maxRegret, p.PNorm, p.CNorm)
+			if err != nil {
+				return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating inferer weight")
+			}
 		}
 		infererWeights[worker] = infererWeight
 	}
 
 	forecasterWeights := make(map[Worker]Weight)
-	for _, worker := range p.Forecasters {
-		forecasterWeight, err := CalcWeightFromNormalizedRegret(normalizedForecasterRegrets[worker], maxRegret, p.PNorm, p.CNorm)
-		if err != nil {
-			return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating forecaster weight")
+	if len(p.ForecasterRegrets) > 0 {
+		for _, worker := range p.Forecasters {
+			var forecasterWeight = alloraMath.ZeroDec()
+			if moreThanOneNotNewInferer && !p.ForecasterRegrets[worker].noPriorRegret {
```

Comment:
for the check of `moreThanOneNotNewInferer` here and above, seems like you can simply skip both loops entirely if it's false, then whenever you check the weight you just check for inclusion before you do calculations with the weight. this fits with the pattern we established of using `palette.[inferers | forecasters]` to guide computation in all steps